tests/unit_tests/test_elasticsearchstore.py

```diff
@@ -229,47 +232,37 @@ def test_set_tag(elastic_run_get_mock, create_store):
-           "status": RunStatus.to_string(RunStatus.RUNNING), "start_time": 1,
-           "lifecycle_stage": LifecycleStage.ACTIVE, "artifact_uri": "artifact_location",
-           "metrics": [m], "params": [p], "tags": [t]}
-    response = [SimpleNamespace(**hit)]
-    search_filter_mock.return_value = Search()
-    search_filter_mock.return_value.execute = mock.MagicMock(return_value=response)
-    real_runs, token = create_store._search_runs(["1"])
-    search_filter_mock.assert_called_once_with("match", experiment_id="1")
-    search_filter_mock.return_value.execute.assert_called_once_with()
-
-    for r in response:
-        metrics = []
-        params = []
-        tags = []
-        for m in r.metrics:
-            metrics.append(ElasticMetric(key=m.key,
-                                         value=m.value,
-                                         timestamp=m.timestamp,
-                                         step=m.step))
-        for p in r.params:
-            params.append(ElasticParam(key=p.key, value=p.value))
-        for t in r.tags:
-            tags.append(ElasticTag(key=t.key, value=t.value))
-        run = ElasticRun(meta={'id': r.meta.id},
-                         experiment_id=r.experiment_id, user_id=r.user_id,
-                         status=r.status,
-                         start_time=r.start_time,
-                         end_time=None,
-                         lifecycle_stage=r.lifecycle_stage, artifact_uri=r.artifact_uri,
-                         metrics=metrics, params=params, tags=tags
-                         )
-    runs = [run.to_mlflow_entity()]
-    assert real_runs[0]._info == runs[0]._info
-    assert real_runs[0]._data._metrics == runs[0]._data._metrics
-    assert real_runs[0]._data._params == runs[0]._data._params
-    assert real_runs[0]._data._tags == runs[0]._data.tags
+def test___build_elasticsearch_query(search_query_mock, create_store):
+    type_dict = ["params", "latest_metrics", "tags"]
+    parsed_filters = [{'type': 'parameter', 'key': 'param0', 'comparator': 'LIKE', 'value': '%va%'},
+                      {'type': 'metric', 'key': 'metric0', 'comparator': '>', 'value': '1'},
+                      {'type': 'tag', 'key': 'tag0', 'comparator': '!=', 'value': 'val2'}]
+    mock_queries = [Q("term", params__key="param0") &
+                    Q('bool', must=[Q("wildcard", params__value="*va*")]),
+                    Q("term", latest_metrics__key="metric0") &
+                    Q('bool', must=[Q("range", latest_metrics__value={'gt': "1"})]),
+                    Q("term", tags__key="tag0") &
+                    Q('bool', must_not=[Q("term", tags__value="val2")])
+                    ]
+    search_query_mock.return_value = Search()
+    for i, parsed_filter in enumerate(parsed_filters):
+        create_store._build_elasticsearch_query(parsed_filters=[parsed_filter], s=Search())
+        search_query_mock.assert_called_with('nested', path=type_dict[i], query=mock_queries[i])
```

Comment:
Instead of creating arrays and looping, you can use `pytest.mark.parametrize`, the advantage is it will create one test by use case, so if one fails we will know directly which one